lib/util/propTypes.js

```diff
@@ -281,6 +281,44 @@ module.exports = function propTypesInstructions(context, components, utils) {
     return ignorePropsValidation;
   }
 
+  function declarePropTypesForTSTypeAnnotation(propTypes, declaredPropTypes) {
+    let members = [];
+
+    if (propTypes.typeAnnotation.type === 'TSTypeReference') {
+      const typeName = propTypes.typeAnnotation.typeName.name;
+      if (!typeName) {
+        return true;
+      }
+      // the game here is to find the type declaration in the code
+      const candidateTypes = context.getSourceCode().ast.body.filter((item) => item.type === 'VariableDeclaration' && item.kind === 'type');
+      const declarations = candidateTypes.map((type) => type.declarations).reduce((acc, val) => acc.concat(val), []);
```

Comment:
```suggestion
      const declarations = flatMap(candidateTypes, (type) => type.declarations);
```
where `flatMap` is `array.prototype.flatmap`?

Reply:
not sure, but weird cases such as type TSFunctionType

```
{
      // neither TSTypeReference or TSTypeLiteral, we do nothing. Weird case
      code: [
        'const Hello = (props: () => any) => {',
        '    return <div>{props.firstname}</div>;',
        '}'
      ].join('\n'),
      parser: parsers.TYPESCRIPT_ESLINT
    }
```
on the other hand, member is empty, so declaredPropsTypes ends up at {}

What is better ? ignoring props validation or returning an empty declaredPropsTypes ?

Reply:
oops, sorry, this is the response for the comment below :)

Reply:
for the flatMap, it needs to be polyfilled for node < 11, and I think it's not the case here. 

It does not work on my computer with node 10 for example

Reply:
Yes, we already have `array.prototype.flatmap` in our deps, so it works down to IE 6 via the import :-)

Reply:
ok cool! trying that asap :)